### ISSUE TITLE
C-u C-u M-. to cause tide-jump-to-definition in new window irrespective or reuse-window variable

### ISSUE THREAD

**Issue by invokesus** [ananthakumaran/tide#255]
## Checklist

- [X] I have searched both open and closed issues and cannot find a duplicate.

I don't think currently there any way to jump to definition` in a new window if `tide-jump-to-definition-reuse-window` is true.

And since the prefix argument is used to jumping to type definition, I would like to add a feature where `C-u C-u M-.` jumps to definition in a new window.

I have `reuse-window` set to `t` since I often need just a glance at the definition but sometimes I need the definition open in a window beside the current one I am working on. This modification will help in that use case.

**Comment by invokesus** [comment]
Another variation could be for `C-u C-u M-.` to perform the opposite behavior of what `reuse-window` variable states.

**Comment by ananthakumaran** [comment]
I will wait for input from others, but my opinion is, this goes into the too specific feature area. Not sure how many people would use it. Also this could be easily solved by

```elisp
(defun my-jump ()
  (let ((tide-jump-to-definition-reuse-window (my-reuse-logic))
    (tide-jump-to-definition))
```

**Comment by josteink** [comment]
I don’t really have an opinion on this either way.

Does other major or minor-modes have any conventions for this kind of navigation and frame management?

**Comment by invokesus** [comment]
[Elpy](https://elpy.readthedocs.io/en/latest/ide.html#navigation) has quite a few.

**Comment by ananthakumaran** [comment]
Should we consider add support for `xref`, then we don't have to worry about the UX and it would be consistent with other packages. Though it's still marked as experimental as of 26.1

```
;; NOTE: The xref API is still experimental and can change in major,
;; backward-incompatible ways.  Everyone is encouraged to try it, and
;; report to us any problems or use cases we hadn't anticipated, by
;; sending an email to emacs-devel, or `M-x report-emacs-bug'.
```

**Comment by ananthakumaran** [comment]
I think in the short term, we can add support for xref without breaking existing functionalities. This means, if xref is available then xref-* functions will automatically start to work. If it works out, then we can consider dropping support for current navigation functionalities later. But all this depends on whether xref api works nicely with our backend etc. I haven't looked at the api closely.

**Comment by josteink** [comment]
Right. Given that everything works across versions and without any need for manual intervention by the user, I'd be happy with such a change.

**Comment by nrxus** [comment]
I would love if `tide` used `xref` to keep it consistent with other modes that use it. This has been a particular pain to find references. I could just remap `tide-references` to the same key as `xref-find-references` but it would be nicer if the UI was also consistent.
